src/main/java/mate/academy/dao/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,48 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.dao.UserDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.User;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    @Override
+    public User add(User user) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(user);

```

Comment:
see difference between `persist` and `save`. 
Use methods accordingly to benefit they provide.
```suggestion
            session.save(user);
```